File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(begin(intervals),end(intervals));
        vector<vector<int>> res;
        for(auto& intr:intervals){
            int a = intr[0];
            int b = intr[1];
            if(!res.size()) res.push_back(intr);
            else{
                vector<int> back = res.back();
                if(back[1]>=a){
                    res.pop_back();
                    a=min(a,back[0]);
                    b=max(b,back[1]);
                }
                res.push_back({a,b});
            }
        }
        return res;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals[:]=sorted(intervals,key=lambda x:x[0])
        res = []
        for interval in intervals: 
            if not(len(res)): 
                res.append(interval)
            else: 
                prev = res[-1] 
                if prev[1]>=interval[0]:
                    res.pop() 
                    new_st = min([prev[0],interval[0]])
                    new_end = max([prev[1],interval[1]])
                    res.append([new_st,new_end])
                else: 
                    res.append(interval)
        
        return res